apps/obs-web/src/views/DebrieferErrors.vue

```diff
@@ -3,79 +3,133 @@
+      uniqueKey="uid"
+      :enableSelection="true"
+      :isFullSize="isFullSize"
+      :loading="loading"
+      @save="save"
+    >
+    </prime-table>
+
   </div>
 </template>
 
-
 <script lang="ts">
-import { mapState } from 'vuex';
-import router from 'vue-router';
-import { State, Action, Getter } from 'vuex-class';
-import { Component, Prop, Vue } from 'vue-property-decorator';
-import {
-  TripState,
-  PermitState,
-  UserState,
-  GeneralState,
-  DebrieferState
-} from '../_store/types/types';
-import {
-  WcgopTrip,
-  WcgopOperation,
-  WcgopCatch,
-  WcgopSpecimen,
-  Basket
-} from '@boatnet/bn-models';
-import { CouchDBCredentials, couchService } from '@boatnet/bn-couch';
-import { Client, CouchDoc, ListOptions } from 'davenport';
-import { date } from 'quasar';
-import { convertToObject } from 'typescript';
-
-@Component
-export default class DebrieferErrors extends Vue {
-  @Action('error', { namespace: 'alert' }) private error: any;
-  @State('debriefer') private debriefer!: DebrieferState;
-  @Prop({ default: true })
-  private showData!: boolean;
-
-  private WcgopTrips: WcgopTrip[] = [];
-  private WcgopOperations: WcgopOperation[] = [];
-  private WcgopCatches: WcgopCatch[] = [];
-  private WcgopCatchSpecies: WcgopCatch[] = [];
-  private WcgopCatchBaskets: Basket[] = [];
-  private WcgopCatchSpecimens: WcgopSpecimen[] = [];
-  private pagination = { rowsPerPage: 50 };
-  private tab = 'trips';
-
-  private errorColumns = [
-    { field: 'severity', header: 'Severity' },
-    { field: 'description', header: 'Description' },
-    { field: 'tripNum', header: 'Trip #' },
-    { field: 'dateCreated', header: 'Date Created' },
-    { field: 'observer', header: 'Observer' },
-    { field: 'status', header: 'Status' },
-    { field: 'dateFixed', header: 'Date Fixed' },
-    { field: 'note', header: 'Note' }
-  ];
-
-  private data = [
-    {
-      severity: 'Warning',
-      description:
-        'Multiple dissections of the same type collected for sea whip, sea pen, or non-coral species.',
-      tripNum: 24266,
-      dateCreated: '1/1/19',
-      observer: 'Davis',
-      status: 'Unresolved',
-      dateFixed: '',
-      note: 'dissection count'
-    }
-  ];
-
-  private openNewDebriefingTab() {
-    const route = '/observer-web/debriefer/qa';
-    window.open(route, '_blank');
+import { createComponent, ref, onMounted, watch, computed } from '@vue/composition-api';
+import moment from 'moment';
+
+import { couchService } from '@boatnet/bn-couch';
+import { Client } from 'davenport';
+import { NoSubstitutionTemplateLiteral } from 'typescript';
+
+export default createComponent({
+  props: {
+    showData: Boolean,
+    isFullSize: Boolean
+  },
+  setup(props, context) {
+    const store = context.root.$store;
+    const state = store.state;
+    const masterDB: Client<any> = couchService.masterDB;
+
+    const errorColumns = [
+      { field: 'severity', header: 'Severity', key: 'errorSeverity', width: '80' },
```

Comment:
this should probably be of type toggle where they can select from a dropdown list of different severities to display. Currently there is just warning and error. Which are probably also good to store as lookups.

Reply:
We should discuss - I was thinking error vs. warning would be set by the rule / error check, and would not be editable, but maybe this field is set by the debriefer taking the broader context into consideration?

If this is just a string set by a function, seems like the options could be baked into the rule rather than lookups, but again, this probably warrants some discussion.

Reply:
oh yes don't want this to be editable agree with that. was just saying for filtering it would be nice to have a dropdown to toggle between error and warning. but yes not for editing.